sql/driver/conn.go

```diff
@@ -29,6 +29,7 @@ import (
 type conn struct {
 	sender  Sender
 	session []byte
+	txn     []byte
```

Comment:
Yes, the session settings are connection specific and manually calling `SET DATABASE` won't work as expected with the Go SQL client due to connection pooling. The usual way this is dealt with via drivers is to have a driver-specific mechanism for setting the database whenever a new connection is opened.

I don't see a particular reason for exposing the transaction here. Simplifying the protobuf and this code will make porting to new languages easier.